pep_sphinx_extensions/tests/pep_processor/transform/test_pep_headers.py

```diff
@@ -0,0 +1,114 @@
+        (
+            "python-announce@python.org",
+            "https://mail.python.org/archives/list/python-announce@python.org/",
+        ),
+    ],
+)
+def test_generate_list_url(test_input, expected):
+    out = pep_headers._generate_list_url(test_input)
+
+    assert out == expected
+
+
+@pytest.mark.parametrize(
+    "test_input, expected",
+    [
+        (
+            "https://mail.python.org/pipermail/python-3000/2006-November/004190.html",
+            ("Python-3000", "message"),
+        ),
+        (
+            "https://mail.python.org/archives/list/python-dev@python.org/thread/HW2NFOEMCVCTAFLBLC3V7MLM6ZNMKP42/",
+            ("Python-Dev", "thread"),
+        ),
+        (
+            "https://mail.python.org/mailman3/lists/capi-sig.python.org/",
+            ("Capi-SIG", "list"),
+        ),
+        (
+            "https://mail.python.org/mailman/listinfo/web-sig",
+            ("Web-SIG", "list"),
+        ),
+        (
+            "https://discuss.python.org/t/pep-643-metadata-for-package-source-distributions/5577",
+            ("Discourse", "thread"),
+        ),
+        (
+            "https://discuss.python.org/c/peps/",
+            ("Peps Discourse", "category"),
+        ),
+    ],
+)
+def test_process_pretty_url(test_input, expected):
+    out = pep_headers._process_pretty_url(test_input)
+
+    assert out == expected
+
+
+@pytest.mark.parametrize(
+    "test_input, expected",
+    [
+        (
+            "https://example.com/",
+            "https://example.com/ not a link to a recognized domain to prettify",
+        ),
+        (
+            "https://mail.python.org",
+            "https://mail.python.org not a link to a list, message or thread",
+        ),
+        (
+            "https://discuss.python.org/",
+            "https://discuss.python.org not a link to a Discourse thread or category",
+        ),
+    ],
+)
+def test_process_pretty_url_invalid(test_input, expected):
+    with pytest.raises(ValueError, match=expected):
+        pep_headers._process_pretty_url(test_input)
+
+
+@pytest.mark.parametrize(
+    "test_input, expected",
+    [
+        (
+            "https://mail.python.org/pipermail/python-3000/2006-November/004190.html",
+            "Python-3000 message",
+        ),
+        (
+            "https://mail.python.org/archives/list/python-dev@python.org/thread/HW2NFOEMCVCTAFLBLC3V7MLM6ZNMKP42/",
+            "Python-Dev thread",
+        ),
+        (
+            "https://mail.python.org/mailman3/lists/capi-sig.python.org/",
+            "Capi-SIG list",
+        ),
+        (
+            "https://mail.python.org/mailman/listinfo/web-sig",
+            "Web-SIG list",
+        ),
+        (
+            "https://discuss.python.org/t/pep-643-metadata-for-package-source-distributions/5577",
+            "Discourse thread",
+        ),
+        (
+            "https://discuss.python.org/c/peps/",
+            "Peps Discourse category",
```

Comment:
Technically speaking, we should do `replace("pep", "PEP")` in the prettifier function to avoid this, like we do for SIG.

Reply:
(But that's my fault)